packages/demo/usecase-demo/src/utils/useCarouselLogic.ts

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+export function useCarouselLogic() {
+  function scrollLeft(carousel: React.RefObject<HTMLDivElement>) {
+    carousel.current?.scrollBy({
+      top: 0,
+      left: -carousel.current.clientWidth,
+      behavior: 'smooth',
+    });
+  }
+  function scrollRight(carousel: React.RefObject<HTMLDivElement>) {
+    carousel.current?.scrollBy({
+      top: 0,
+      left: carousel.current.clientWidth,
+      behavior: 'smooth',
+    });
+  }
+  return {
+    scrollLeft,
+    scrollRight,
+  };
+}
```

Comment:
Only used by `CarouselScrollArrows`.
Not sure if it's worth it to have a separate hook here. I'd suggest to move these 2 functions directly to `CarouselScrollArrows.tsx` component.